src/Docker.PowerShell/Support/MultiplexedStreamExtensions.cs

```diff
@@ -4,18 +4,43 @@
 using System.Threading.Tasks;
 using Docker.DotNet;
 
+#if !NET46
+using System.Runtime.InteropServices;
+#endif
+
 namespace Docker.PowerShell.Support
 {
     internal static class MultiplexedStreamExtensions
     {
+#if NET46
+        private static readonly bool IsWindows = true;
+        private static readonly bool IsLinux = false;
+#else
+        private static readonly bool IsWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        private static readonly bool IsLinux = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+#endif
+
         public static async Task CopyToConsoleAsync(this MultiplexedStream stream, bool tty, bool openStdin, CancellationToken cancellationToken)
         {
             Stream stdin = Stream.Null, stdout = Stream.Null, stderr = Stream.Null;
             ConsoleStream conin = null, conout = null;
+            Task stdinReadTask = null;
             try
             {
                 // TODO: What if we are not attached to a console? If config.Tty is false, this should not be an error.
-                conout = new ConsoleStream(ConsoleDirection.Out);
+
+                if (IsWindows)
+                {
+                    conout = new Win32ConsoleStream(ConsoleDirection.Out);
+                }
+                else if (IsLinux)
```

Comment:
Currently the Linux behavior is generic, so maybe we should just default to that if not on Windows.